plenum/common/messages/internal_messages.py

```diff
@@ -38,6 +39,14 @@
                              ('dst', List[str]),
                              ('stash_data', Optional[tuple])])
 
+# TODO: This should be merged with RaisedSuspicion
+VoteForViewChange = NamedTuple('VoteForViewChange', [('suspicion', Suspicion)])
+
+# TODO: This is a kind of hack to make Node process NeedViewChange before replicas
+#  Possible solution to this (and some other hacks) would be adding separate node internal bus
+#  to which such messages are sent, processed, and then forwarded to replicas buses if needed
+PreNeedViewChange = NamedTuple('PreNeedViewChange', [('view_no', int)])
```

Comment:
Not sure. Probably better solution would be renaming PreNeedViewChange to NodeNeedViewChange (instead of PreNeedViewChange) and either keeping NeedViewChange as is or renaming it to ReplicaNeedViewChange. However introducing separate Node internal bus, from which messages are forwarded to Replicas internal buses looks like a clearer (but not cheaper) option for me, since it would not require introducing separate message at all. Any thoughts?

Reply:
Yes, I would prefer to avoid having multiple messages doing the same, so a separate bus is a better solution I think. But if two messages are kept for a while, then I would prefer not to use 'PreViewChange...' name.